src/Angular.js

```diff
@@ -125,6 +127,30 @@ var VALIDITY_STATE_PROPERTY = 'validity';
 
 var hasOwnProperty = Object.prototype.hasOwnProperty;
 
+var objectMaxDepthInErrorMessage = 5;
+
+/**
+ * @ngdoc function
+ * @name angular.configureErrorHandling
+ * @module ng
+ * @kind function
+ *
+ * @description Use this function to configure minErr
+ *
+ * @param {Object=} config an object for defining configuration for error handling. The
+ *     following keys are supported:
+ *
+ * * `objectMaxDepth` - used to stringify objects in error messages until reaching the max depth.
+ *
+ */
+function configureErrorHandling(config) {
+  if (isObject(config)) {
+    if (isNumber(config.objectMaxDepth) && config.objectMaxDepth > 0) {
```

Comment:
It is now impossible to unset it (which means that there will always be copying in `toDebugString`). I would change this to something like:

```js
objectMaxDepthInErrorMessage = isValidObjectMaxDepth(config.objectMaxDepth) ? config.objectMaxDepth : NaN;

function isValidObjectMaxDepth(maxDepth) {
  return isNumber(maxDepth) && maxDepth > 0;
}
```

<sub>And you can reuse `isValidObjectMaxDepth()` inside `copy()` and `serializeObject()` for DRY-ness and consistency.
</sub>